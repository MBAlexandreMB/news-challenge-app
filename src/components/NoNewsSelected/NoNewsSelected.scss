@import '../../shared/styles/sizes.scss';

.no-news-container {
    display: flex;
    flex-flow: column nowrap;
    align-items: center;
    position: sticky;
    top: 10px;

    width: calc(100% - 437px);
    height: fit-content;

    .no-news-image {
        width: 25vw;
    
        img {
            width: 100%;
        }
    }

    h1 {
        font-size: 2vw;
        text-align: center;
    }
}

@media (max-width: #{$breakpoint-tablet}) {
    .no-news-container {
        display: none;
    }
}