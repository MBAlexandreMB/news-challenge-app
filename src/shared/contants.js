export const NO_IMAGE_URL = "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRBJvWZNK1Lm7z-Iw0g4RJvw6l1sPwOUDOaZW_AecE3ZZdPyy4KpWejikv7VSD4cuUQpOY&usqp=CAU";