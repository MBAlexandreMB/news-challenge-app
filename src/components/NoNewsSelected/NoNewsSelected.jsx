import { Fragment } from 'react';
import { NO_NEWS_SELECTED_TEXT } from '../../shared/contants';

import Loader from '../../shared/icons/Loader';
import './NoNewsSelected.scss';

const NoNewsSelected = ({isLoading}) => {
    return (
        <section className='no-news-container'>
            { isLoading
            ? <Loader />
            : (
                <Fragment>
                    <figure className="no-news-image">
                        <img src={process.env.PUBLIC_URL + '/images/newspaper.png'} alt='Newspaper' />
                    </figure>
                    <h1>{NO_NEWS_SELECTED_TEXT}</h1>
                </Fragment>
            )}
        </section>
    );
}
 
export default NoNewsSelected;