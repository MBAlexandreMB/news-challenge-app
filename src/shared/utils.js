import { NO_IMAGE_URL } from './contants';

export const onImageLoadingError = ({ currentTarget }) => {
    currentTarget.onerror = null;
    if (currentTarget.src !== NO_IMAGE_URL) {
        currentTarget.src= NO_IMAGE_URL;
    }
};