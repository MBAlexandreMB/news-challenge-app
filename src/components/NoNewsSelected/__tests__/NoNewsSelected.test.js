import { shallow } from 'enzyme';
import { NO_NEWS_SELECTED_TEXT } from '../../../shared/contants';
import NoNewsSelected from '../NoNewsSelected';

describe('<NoNewsSelected />', () => {
    it('should render correctly without props', () => {
        const wrapper = shallow(<NoNewsSelected />);

        expect(wrapper).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render correctly when loading', () => {
        const wrapper = shallow(<NoNewsSelected isLoading={true} />);
        const loader = wrapper.find('Loader');

        expect(loader.exists()).toBeTruthy();
        expect(wrapper).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render correctly when NOT loading', () => {
        const wrapper = shallow(<NoNewsSelected isLoading={false} />);
        const loader = wrapper.find('Loader');
        const figure = wrapper.find('figure');
        const text = wrapper.find('h1').text();

        expect(loader.exists()).toBeFalsy();
        expect(figure.exists()).toBeTruthy();
        expect(text).toBe(NO_NEWS_SELECTED_TEXT);
        expect(wrapper).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });
});