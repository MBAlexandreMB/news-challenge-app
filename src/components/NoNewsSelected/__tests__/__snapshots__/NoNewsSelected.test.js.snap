// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoNewsSelected /> should render correctly when NOT loading 1`] = `
<section
  className="no-news-container"
>
  <figure
    className="no-news-image"
  >
    <img
      alt="Newspaper"
      src="/images/newspaper.png"
    />
  </figure>
  <h1>
    Select something on the left list
  </h1>
</section>
`;

exports[`<NoNewsSelected /> should render correctly when loading 1`] = `
<section
  className="no-news-container"
>
  <Loader />
</section>
`;

exports[`<NoNewsSelected /> should render correctly without props 1`] = `
<section
  className="no-news-container"
>
  <figure
    className="no-news-image"
  >
    <img
      alt="Newspaper"
      src="/images/newspaper.png"
    />
  </figure>
  <h1>
    Select something on the left list
  </h1>
</section>
`;
